sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,228 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#==============================================================================
+
+from __future__ import division
+from math import log
+import numpy as np
+from scipy.stats import entropy
+from sklearn.base import BaseEstimator
+
+class MDLP(BaseEstimator):
+    """Implements the MDLP discretization criterion from Usama Fayyad's
+    paper "Multi-Interval Discretization of Continuous-Valued Attributes
+    for Classification Learning"
+    """
+
+    def __init__(self, **params):
+        self.set_params(**params)
+
+    def get_params(self, deep=True):
+        return {"continuous_columns": self.continuous_columns,
+                "min_depth": self.min_depth}
+
+    def set_params(self, **params):
+        self.continuous_columns = params.get("continous_columns")
+        self.min_depth = params.get("min_depth", 0)
+        return self
+
+    def fit(self, X, y):
+        """Finds the intervals of interest from the input data.
+        """
+        if type(X) is list:
+            X = np.array(X)
+        if type(y) is list:
+            y = np.array(y)
+        if self.continuous_columns is None:
+            self.continuous_columns = range(X.shape[1])
+        if(len(X.shape) != 2):
+            raise ValueError("MDLP can ony be applied to input ndarrays of "
+                             "size 2.")
+        if(X.shape[0] != y.shape[0]):
+            raise ValueError("Number of samples in X does not match number "
+                             "of targets in y")
+
+        self.num_classes_ = set(y)
+        self.intervals_ = dict()
```

Comment:
I imagine a column to be associated with a specific list of intervals. Which is why I wanted `self.intervals_` to be a dictionary; it represents a mapping from a column to a list of intervals. 

Furthermore, I wanted to use a dictionary, because this makes its compatibility with pandas `DataFrame` objects easier in the future. A pandas `Series` (indexed by a string) would map to their corresponding list of intervals. (But obviously, we need to go one step at a time, and work on this pull request first.)

Does that sound fair? This implementation sounds intuitive to me at least.